src/virtual-machine.js

```diff
@@ -284,6 +284,20 @@ class VirtualMachine extends EventEmitter {
     }
 
     /**
+     * Delete a costume from the current editing target.
+     * @param {int} costumeIndex - the index of the costume to be removed.
+     */
+    deleteCostume (costumeIndex) {
+        this.editingTarget.sprite.costumes = this.editingTarget.sprite.costumes
+            .slice(0, costumeIndex)
+            .concat(this.editingTarget.sprite.costumes.slice(costumeIndex + 1));
+        if (costumeIndex === this.editingTarget.currentCostume) {
+            this.editingTarget.setCostume(costumeIndex - 1);
+        }
```

Comment:
And jumping off of my comment on https://github.com/LLK/scratch-vm/pull/568, can we move this logic to the `RenderedTarget`? The target should maintain its own asset lists.